src/main/scala/cognite/spark/v1/SequenceRowsRelation.scala

```diff
@@ -157,28 +161,39 @@ class SequenceRowsRelation(config: RelationConfig, sequenceId: CogniteId)(val sq
           (index, field.name, columnType)
       }
 
-    def parseRow(row: Row): SequenceRowWithExternalId = {
+    def parseRow(row: Row): SequenceRowWithId = {
       val rowNumber = row.get(rowNumberIndex) match {
         case x: Long => x
         case x: Int => x: Long
         case _ => throw SparkSchemaHelperRuntime.badRowError(row, "rowNumber", "Long", "")
       }
 
-      val externalId = row.get(externalIdIndex) match {
-        case x: String => x
-        case _ => throw SparkSchemaHelperRuntime.badRowError(row, "externalId", "String", "")
+      val maybeExternalId = Try(row.get(externalIdIndex)).toOption match {
```

Comment:
`row.get` will always succeed if the `externalIdIndex` is >= 0, I'd much prefer an if check on the index instead of throwing and catching an exception